dev/testsreporter/testsreporter.go

```diff
@@ -30,6 +31,8 @@ type CheckOptions struct {
 }
 
 func Check(resultsPath string, options CheckOptions) error {
+	ctx := context.TODO()
```

Comment:
Mage targets can have a context as first argument, so this could be defined as 
```suggestion
func Check(ctx context.Context, resultsPath string, options CheckOptions) error {
```

And the target that calls it:
```go
func ReportFailedTests(ctx context.Context, testResultsFolder string) error {
```

Reply:
Thanks! Updated here  https://github.com/elastic/integrations/pull/12408/commits/164734030af12a3de92872178f353f0e3e438c2b